### PR TITLE
feat: Product module events

### PR DESCRIPTION
**Pull request by adrien2p**
WIP, not ready to review

**what**
- add events
- integration tests of emitted events
- remove integration tests on auto generated services in favor of module method integration tests

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: packages/modules/product/src/types/index.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ export type InitializeModuleInjectableDependencies = {
   eventBusModuleService?: IEventBusModuleService
 }
 
+// TODO: remove and cleanup bellow code
```
Comment (sradevski) [inline packages/modules/product/src/types/index.ts:8]: Is there anything stopping us from doing it right away?

**Review by sradevski — APPROVED** [review]

**Comment by olivermrbl** [comment]
@adrien2p, can we merge this?

**Comment by adrien2p** [comment]
@olivermrbl no it is not ready, I ve marked it as WIP not ready in the description ahah 😓

**Comment by olivermrbl** [comment]
Got it, I just saw you requested a review yesterday and two approvals 😄

**Comment by adrien2p** [comment]
> Got it, I just saw you requested a review yesterday and two approvals 😄

Yeah I clicked on it by mistake and added the desc quicky 😂 my bad for the false alarm ahah

**Comment by adrien2p** [comment]
@olivermrbl maybe we can already merge this as a first iteration for now and worst case open another pr to continue. I sugges to re review just in case

**Review by sradevski — APPROVED** [review]
LGTM, nice cleanup with the tests

**Pull request merged**